JiraReporter/Services/AuthorLoader.cs

```diff
@@ -391,19 +396,39 @@ private Image GetImageFromURL(string url)
                 EncryptionService = EncryptionService
             }.Authorize(webClient,_context.JiraRequestContext, UrlExtensions.GetRelativeUrl(url));
 
-            var imageData = webClient.DownloadData(url);
+            return webClient.DownloadData(url);
+        }
+
+        private void SetUserAvatar(byte[] image)
+        {
+            using(var db = new ReportsDb())
+            {
+                var installedInstace = db.InstalledInstances.Single(i => i.Id == _context.Settings.InstalledInstance.Id);
+
+                if (!installedInstace.UserImages.ToList().Exists(userImage => userImage.Username == _currentAuthor.Username))
+                    SaveAvatarToDb(image, installedInstace);
 
-            MemoryStream stream = new MemoryStream(imageData);
-            var img = Image.FromStream(stream);
-            stream.Close();
+                var authorAvatar = installedInstace.UserImages.Single(a=>a.Username == _currentAuthor.Username);
```

Comment:
Exact cum am spus mai sus, poti sa pui linia asta inainte if-ului si cu SingleOrDefault in loc de Single. 

Daca nu exista atunci adaugi.